cmd/fetch/fetch.go

```diff
@@ -64,6 +64,9 @@ func fetch(cmd *cobra.Command, args []string) error {
 
 func fetchConnection(ctx context.Context, specReader *specs.SpecReader, connSpec specs.ConnectionSpec, pm *plugin.PluginManager) error {
 	sourceSpec := specReader.GetSourceByName(connSpec.Source)
+	if sourceSpec.Name == "" {
```

Comment:
can we move this directly to unmarshal code? that this is required. This way we wont need to do this additional check everywhere in the code?

Reply:
this is a check to see if `connSpec.Source` was found in the specreader or not. moving it to some other place won't do any good. Other option is to make the `SpecReader.Get<T>ByName` methods return `<T>, bool` or `<T>, error`.

Reply:
What we could do is make sure it always finds the source, by checking every conn spec against the list of sources? That's more organic and would enable us to skip that check, but if it fails for some other reason, we'll still get the unclear errors.